docs/testing/tips.md

```diff
@@ -0,0 +1,251 @@
+
+For single-cell range, use the singular versions of the API. For multi-cell range or for a range that you don't know in advance how many cells it may contain, use the plural versions of the API.
+
+You may wish to convert the 2-dimensional result into a 1-dimensional array for further processing. Let's see how to achieve that for a single row and single column.
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook)
+{
+  const table = workbook.getActiveWorksheet().getTables()[0];
+  const header = table.getHeaderRowRange().getTexts();
+  // Display header as 1D array - extract 1st row.
+  console.log(header[0]);
+
+
+  const sales = table.getColumnByName('Sales').getRange().getValues();
+  console.log(sales);
+  
+  // Get column-0.
+  const salesAs1DArray = extractColumn(sales, 0);
+  console.log(salesAs1DArray);
+
+  // Add 100 to each value.
+  const revisedSales = salesAs1DArray.map(data => data as number + 100);
+  // Add a column.
+  table.addColumn(-1, revisedSales);
+
+  const salesBackTo2D = convertColumnTo2D(salesAs1DArray);
+  console.log(salesBackTo2D);
+
+}
+
+/**
+ * Extract a column from 2D array and return result.
+ */
+function extractColumn(data: (string | number | boolean)[][], index: number): (string | number | boolean)[] {
+
+  const column = data.map((row) => {
+    return row[index];
+  })
+  return column;
+}
+
+/**
+ * Convert a flat array into 2D array that can be used as range column.
+ */
+function convertColumnTo2D(data: (string | number | boolean)[]): (string | number | boolean)[][] {
+
+  const columnAs2D = data.map((row) => {
+    return [row];
+  })
+  return columnAs2D;
+}
+```
+
+For setter APIs, the model is similar. There are singular `setValue()`, `setFormula()`, etc. APIs that accept an argument of a single type (string or number or boolean depending on the API). If you use these methods on a range with multiple cells, all cells will be set to the same value/formula, etc.
+
+* `range.setValue('Hello')` --> this will set all cells in the range to value `Hello`.
+* `range.setValues([['Hello', 'World!]])` --> this will set two adjacent cells of the range. If you are using a plural version of the API, the dimension of the range should match the dimension of the argument.
+
+## Other miscellaneous tips
+
+## Full screen editing
+
+A full-screen detachable editor is not yet offered in Office Scripts. In the meantime, you can use the following steps to get code completion in Visual Studio Code. It's only for editing; you can't execute a script using this. To run the script, you still have to copy/paste code into the Office Scripts in-built code editor.
+
+1. In case you already don't have it, download and install the [Visual Studio Code editor](https://code.visualstudio.com/).
+1. Download the intelliSense (code completion assistance) [file](index.d.ts) to your local machine into a base folder from where you'll edit your script files.
```

Comment:
Where should we grab index.d.ts from? Or should we have user just point directly to DT or office-scripts copy?